src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,180 @@
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            var delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol.Equals(delegateCreationOperation.Type))
+                            {
+                                var alwaysReturnTrue = false;
+                                var kindOfTargetFunction = delegateCreationOperation.Target.Kind;
+
+                                switch (kindOfTargetFunction)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        alwaysReturnTrue = AlwaysReturnTrue(delegateCreationOperation.Target.Descendants());
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        var methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        var methodSymbol = methodReferenceOperation.Method;
+                                        var blockOperation = methodSymbol.GetTopmostOperationBlock(compilationStartAnalysisContext.Compilation);
+                                        // TODO(LINCHE): This is an issue tracked by #2009. We filter extraneous based on IsImplicit.
+                                        var targetOperations = FilterImplicitOperations(ImmutableArray.ToImmutableArray(blockOperation.Descendants()));
+                                        alwaysReturnTrue = AlwaysReturnTrue(targetOperations);
+                                        break;
+                                }
+
+                                if (alwaysReturnTrue)
+                                {
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        delegateCreationOperation.CreateDiagnostic(
+                                            Rule,
+                                            kindOfTargetFunction.ToString()));
+                                }
+                            }
+                        },
+                        OperationKind.DelegateCreation);
+                });
+        }
+
+        /// <summary>
+        /// Gets all valid members of the block operation body, excluding the VB implicit statements.
+        /// </summary>
+        /// <param name="operations">All the descendants of the IBlockOperation of target method.</param>
+        private static ImmutableArray<IOperation> FilterImplicitOperations(ImmutableArray<IOperation> operations)
```

Comment:
However, when the test code is ServicePointManager.ServerCertificateValidationCallback += (a, b, c, d) => true; the IReturnOperation of "true" will be Implicit. If implement in this way, it will filter this IReturnOperation. Besides, I don't know if there's any other Implicit situation. So, I adopt a safer way.

Reply:
Ah, actually, what you need is just to port this method [WithoutFullyImplicitOperations](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Utilities/Extensions/IOperationExtensions.cs#L142) from features/dataflow branch into master - that helper will essentially filter out all the compiler generated operation trees. I think then you then don't need any workaround inside the analyzer, just invoking this helper will give you needed operations. Can you give it a try and see if it works?

Reply:
Did you try the suggestion above to port WithoutFullyImplicitOperations from features/dataflow branch into master? Seems like your current code is still filtering out all the implicit operations, which will not handle the expression body case that you mentioned above?